### DISCUSSION TITLE
Migrating from Eventlet

### DISCUSSION THREAD

**Discussion by vpmedia** [miguelgrinberg/Flask-SocketIO#2027] [Q&A]
Reading the following issue in the Eventlet page, the question arises in myself that to what concurrency framework should I plan to migrate?
https://github.com/eventlet/eventlet/issues/824

gevent? or are there any other option which is suitable for production? is async.io supported?

thanks!

**Comment by miguelgrinberg** [comment]
Gevent should work fine, with little or no changes to your application.

The python-socketio project also supports asyncio, but Flask and Flask-SocketIO do not.

**Comment by vpmedia** [comment]
@miguelgrinberg If I need WebSocket support I see the recommendation is to use `gevent-websocket` (last release 12th Marc 2017) or `uwsgi` (in maintenance mode). So I'm not entirely confident that these would be good choices if I'm planning the foundations (used frameworks) of my new project since neither of these seem to be under active development.

**Reply by miguelgrinberg** [reply]
You do not need gevent-websocket. I actually have to update the documentation to make it more clear that gevent is currently the only dependency needed. You can use gevent-websocket if you like, but if this package isn't installed then WebSocket support is provided by simple-websocket which is installed as a dependency.

**Reply by vpmedia** [reply]
Great! Thanks for the response!
